pkg/registry/apiserverinternal/storageversion/storage/storage.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/apis/apiserverinternal"
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/printers"
+	printersinternal "k8s.io/kubernetes/pkg/printers/internalversion"
+	printerstorage "k8s.io/kubernetes/pkg/printers/storage"
+	strategy "k8s.io/kubernetes/pkg/registry/apiserverinternal/storageversion"
+)
+
+// REST implements a RESTStorage for storage version against etcd
+type REST struct {
+	*genericregistry.Store
+}
+
+// NewREST returns a RESTStorage object that will work against storageVersions
+func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, error) {
```

Comment:
In order to have separated spec and status (and we should keep to the convention).  We need to have separate statusStore.  See https://github.com/kubernetes/kubernetes/blob/87b2a3129b57adf8872c4cc0ddc43646c23fe9b3/pkg/registry/apps/daemonset/storage/storage.go#L59 for an example.